Kudu.Services/Deployment/DeploymentController.cs

```diff
@@ -392,6 +399,37 @@ public HttpResponseMessage GetResult(string id)
             }
         }
 
+        private bool IsLatestPendingDeployment(ref string id)
+        {
+            if (String.Equals(Constants.LatestDeployment, id))
+            {
+                using (_tracer.Step("DeploymentService.GetLatestDeployment"))
+                {
+                    var results = _deploymentManager.GetResults();
+                    var pending = results.Where(r => r.Status != DeployStatus.Success && r.Status != DeployStatus.Failed).FirstOrDefault();
+                    if (pending != null)
+                    {
+                        _tracer.Trace("Deployment {0} is {1}", pending.Id, pending.Status);
+                        return true;
+                    }
+
+                    var latest = results.Where(r => r.EndTime != null).OrderBy(r => r.EndTime.Value).LastOrDefault();
```

Comment:
is the EndTime null check necessary?
after above pending deployment check, there shouldn`t be any deployment that without "EndTime" right?